### PR TITLE
Remove GAP stones

### PR DESCRIPTION
**Pull request by fingolfin**
This implements issue #1060.

However, it is not quite ready yet, because I did nothing to adapt `testtravis`, which previously run a subset of testinstall and teststandard, selecting only those .tst files whoe GAPstones were below a certain threshold.

I think we should change testtravis to not run testinstall (as we run that in a separate travis task already anyway). Next, we should identify which teststandard tasks take particularly long, and then either make them faster, or else consider moving them into a yet another category, say "testslow" or "testextensive" or so.

### CODE REVIEW

**Comment by fingolfin** [comment]
OK, I now added a bunch of tweaks that go beyond removing stones, but which try to get a better working "testravis" target. Let's see how that works out.

For this, I moved some of the slowest .tst files to a new directory `testextra`, which is not yet hooked by anything. I could of course add a `tst/testextra.g` file to run them, but I thought I'd first see whether (a) this works as I hope it will work, and (b) whether anybody hates the idea...

BTW, I think the reason Travis kills our slow tests is not that they take long, but rather that there is no output for more than 10 minutes. So if we split them into multiple quicker files, or enabled the `showProgress` option, we likely could run them anyway. The latter would be a bit annoying as we'd get lots of unhelpful output, though.

**Comment by ChrisJefferson** [comment]
I am strongly in favour of splitting up test files into smaller pieces.

Just an advanced warning. I tried this with bugfix.tst and found it broke other tests, as pieces were run earlier, and messed up bits of gap state. Could move bugfix.tst also to its own directory so it can be split up of course.

**Comment by fingolfin** [comment]
Awesome, those test failures are due to a genuine bug :-)

**Comment by fingolfin** [comment]
Rebased version, now builds on PR #1100. Also removed the "do not merge" label -- from my point of view, this is ready.

File: tst/testinstall/xgap.tst [hunk 1]
```diff
@@ -53,10 +51,6 @@ gap> HasSize(t);
 true
 gap> Size(t); 
 1
-gap> f := FreeGroup(2);;
-gap> g:=f/[f.1^2,f.2^3];;
-gap> g.1^5=g.1;
-true
 gap> STOP_TEST( "xgap.tst", 62690000);
```
Comment (alex-konovalov) [inline tst/testinstall/xgap.tst]: Any special reasons to remove this?

Comment (fingolfin) [inline tst/testinstall/xgap.tst]: It took several seconds (5-10?) to complete, so does not belong into `testinstall`. 

I contemplated moving it to a new test file in `teststandard`, but ultimately decided that this is a weird random test, for whch it is unclear what exactly it is meant to exercise. Actually, the whole `xgap.tst` somewhat falls into that category. From the original commit by Max N., who added it 1999-05-30: "Added tests for fp groups for problems found with XGAP. Max."

So ultimately, this file should go, its contents (or equivalent content) distributed over more appropriate files (e.g. `fpgrps.tst` or files in `opers`).

**Comment by fingolfin** [comment]
@alex-konovalov As to adapting scripts which parse the output of `TestDirectory`: That sounds quite brittle. Were are those scripts? What are they used for? Perhaps we can find a better solution than parsing output meant for humans, which will always be brittle.

**Comment by fingolfin** [comment]
@alex-konovalov Also not PR #1100, which is a subset of this PR. We could get that in first (with suitable changes), as that should not break any of your scripts.

**Pull request merged**